cmd/prometheus/main.go

```diff
@@ -428,7 +428,11 @@ func main() {
 	// Throw error for invalid config before starting other components.
 	var cfgFile *config.Config
 	if cfgFile, err = config.LoadFile(cfg.configFile, agentMode, false, log.NewNopLogger()); err != nil {
-		level.Error(logger).Log("msg", fmt.Sprintf("Error loading config (--config.file=%s)", cfg.configFile), "err", err)
+		absPath, err := filepath.Abs(cfg.configFile)
+		if err != nil {
```

Comment:
```suggestion
		if pathErr != nil {
```

Otherwise we would hide the real error from the user.

Reply:
Done.